gluster/block/cmd/glusterblock-provisioner/glusterblock-provisioner.go

```diff
@@ -204,13 +214,19 @@ func (p *glusterBlockProvisioner) Provision(options controller.VolumeOptions) (*
 		iscsiVol.Iqn = blockVol.heketiBlockVolRes.Iqn
 		iscsiVol.User = blockVol.heketiBlockVolRes.User
 		iscsiVol.AuthKey = blockVol.heketiBlockVolRes.AuthKey
-		iscsiVol.BlockVolName = blockVolPrefix + blockVol.heketiBlockVolRes.ID
+		if cfg.volumeNamePrefix != "" {
+			iscsiVol.BlockVolName = blockVolName
+		} else {
+			iscsiVol.BlockVolName = blockVolPrefix + blockVol.heketiBlockVolRes.ID
+		}
+		iscsiVol.VolumeID = blockVol.heketiBlockVolRes.ID
 	} else if cfg.opMode == glusterBlockOpmode && blockVol.glusterBlockExecVolRes != nil {
 		iscsiVol.Portals = blockVol.glusterBlockExecVolRes.Portals
 		iscsiVol.Iqn = blockVol.glusterBlockExecVolRes.Iqn
 		iscsiVol.User = blockVol.glusterBlockExecVolRes.User
 		iscsiVol.AuthKey = blockVol.glusterBlockExecVolRes.AuthKey
 		iscsiVol.BlockVolName = blockVolName
+		iscsiVol.VolumeID = ""
```

Comment:
why empty?

Reply:
@rootfs in gluster-block mode, this volID has no significance, so the annotation has set to ""